### PR TITLE
feat: Add GMP destination type

### PR DESCRIPTION
**Pull request by dehaansa**
Adds the ability to configure a GMP destination type.

This is needed for an ongoing project validating GMP support through otel with BPOP, but in the long run we will want to modify the exporter similar to the GCP exporter to support explicit credential configuration.

image

image

### CODE REVIEW

**Comment by dehaansa** [comment]
Updated as requested to show prometheus logo 
image

File: resources/destination-types/google_managed_prometheus.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: bindplane.observiq.com/v1
+kind: DestinationType
+metadata:
+  name: googlemanagedprometheus
+  displayName: Google Managed Prometheus
+  icon: /icons/destinations/prometheus.svg
+  description: Send metrics to Google Managed Prometheus.
+spec:
+  parameters:
```
Comment (cpheps) [inline resources/destination-types/google_managed_prometheus.yaml:9]: can you add a version field with version `0.0.1`?

Comment (dehaansa) [reply resources/destination-types/google_managed_prometheus.yaml:10]: Added.

**Review by cpheps — APPROVED** [review]

**Comment by BinaryFissionGames** [comment]
I've updated this PR with credential options; 

Contingent on this PR on the collector to work: https://github.com/observIQ/observiq-otel-collector/pull/1035

Screen Shot 2023-04-20 at 3 30 05 PM

**Comment by jsirianni** [comment]
I added an "advanced" option for specifying a default `location` resource attribute. Location is the only required attribute,  so we should be setting it to a sane default if it does not already exist.

**Comment by jsirianni** [comment]
@cpheps this is ready to go now. I added an enum for "default location". PR stack is configured with host metrics.

Screenshot from 2023-04-21 14-41-18

**Review by cpheps — APPROVED** [review]
LGTM

**Review by jsirianni — APPROVED** [review]

**Pull request merged**